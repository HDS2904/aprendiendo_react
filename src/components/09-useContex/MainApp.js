import React from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'


export const MainApp = () => {
	const user = {
		id: 120,
		name: 'Jonathan',
		email: 'jonathan.canales.hds@gmail.com',
	}
	return (
		<UserContext.Provider value={user} >
			<AppRouter/>
		</UserContext.Provider>
	)
}
