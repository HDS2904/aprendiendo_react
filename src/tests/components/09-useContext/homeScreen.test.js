import { render, screen } from "@testing-library/react"
import { HomeScreen } from "../../../components/09-useContex/HomeScreen"
import { UserContext } from "../../../components/09-useContex/UserContext"

describe('Pruebas en <HomeScreen />', () => {

	let wrapper;
	const user = {
		name: 'Jonathan',
		email: 'jonathan.canales.hds@gmail.com'
	}

	beforeEach(() => {
		wrapper = render(
			<UserContext.Provider value={{
				user
			}}
			>
				<HomeScreen />
			</UserContext.Provider>
		)
	})

	test('Debe mostrarse correctamente', () => {

		expect( wrapper.container ).toMatchSnapshot()

	})

})