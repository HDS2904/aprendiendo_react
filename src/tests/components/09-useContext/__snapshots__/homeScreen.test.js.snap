// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> Debe mostrarse correctamente 1`] = `
<div>
  <div>
    <h1>
      HomeScreen
    </h1>
    <hr />
    <pre>
      {
   "name": "Jonathan",
   "email": "jonathan.canales.hds@gmail.com"
}
    </pre>
  </div>
</div>
`;
